tool/src/cpp/docs.rs

```diff
@@ -0,0 +1,239 @@
+use std::fmt::Write;
+use std::{collections::HashMap, fmt};
+
+use diplomat_core::ast;
+use indenter::indented;
+
+use crate::{
+    cpp::{structs::gen_method_interface, types::gen_type},
+    docs_util::markdown_to_rst,
+};
+
+/// Generate RST-formatted Sphinx docs for all FFI types.
+pub fn gen_docs(
+    env: &HashMap<ast::Path, HashMap<String, ast::ModSymbol>>,
+    outs: &mut HashMap<String, String>,
+) -> fmt::Result {
+    let index_out = outs
+        .entry("index.rst".to_string())
+        .or_insert_with(String::new);
+    writeln!(index_out, "Documentation")?;
+    writeln!(index_out, "=============")?;
+    writeln!(index_out)?;
+    writeln!(index_out, ".. toctree::")?;
+    let mut toctree_indent = indented(index_out).with_str("   ");
+    writeln!(&mut toctree_indent, ":maxdepth: 3")?;
+    writeln!(&mut toctree_indent, ":caption: Modules:")?;
+    writeln!(&mut toctree_indent)?;
+    let mut sorted_keys: Vec<String> = env
```

Comment:
We really should provide the ability to name our modules

Reply:
#66